src/api/mod.rs

```diff
@@ -745,38 +790,83 @@ impl<T: Pixel> Context<T> {
-  /// until twopass_in() returns 0.
+  /// Returns the number of bytes of the stats file needed before the next
+  /// frame of the second pass in a two-pass encode can be encoded.
+  ///
+  /// This is a lower bound (more might be required), but if `0` is returned,
+  /// then encoding can proceed. This is just a hint to the application, and
+  /// does not need to be called for encoding the second pass to work, so long
+  /// as the application continues to provide more data to [`twopass_in`] in a
+  /// loop until [`twopass_in`] returns `0`.
+  ///
+  /// [`twopass_in`]: #method.twopass_in
+  #[inline]
   pub fn twopass_bytes_needed(&mut self) -> usize {
     self.inner.rc_state.twopass_in(None).unwrap_or(0)
   }
 
-  /// Provide stats data produced in the first pass of a two-pass encode to the
-  /// second pass. On success this returns the number of bytes of that data
-  /// which were consumed. When encoding the second pass of a two-pass encode,
-  /// this should be called repeatedly in a loop before every call to
-  /// receive_packet() (including the very first one) until no bytes are
-  /// consumed, or until twopass_bytes_needed() returns 0.
+  /// Provides the stats data produced in the first pass of a two-pass encode
+  /// to the second pass.
+  ///
+  /// On success this returns the number of bytes of the data which were
+  /// consumed. When encoding the second pass of a two-pass encode, this should
+  /// be called repeatedly in a loop before every call to [`receive_packet`]
+  /// (including the very first one) until no bytes are consumed, or until
+  /// [`twopass_bytes_needed`] returns `0`.
+  ///
+  /// [`receive_packet`]: #method.receive_packet
+  /// [`twopass_bytes_needed`]: #method.twopass_bytes_needed
+  #[inline]
   pub fn twopass_in(&mut self, buf: &[u8]) -> Result<usize, EncoderStatus> {
     self.inner.rc_state.twopass_in(Some(buf)).or(Err(EncoderStatus::Failure))
   }
 
+  /// Returns the encoded data.
+  ///
+  /// # Examples
+  ///
+  /// ``` no_run
+  /// use rav1e::prelude::*;
+  /// #
+  /// # fn main() -> Result<(), EncoderStatus> {
+  /// let cfg = Config::default();
+  /// let mut ctx: Context<u8> = cfg.new_context();
+  /// let frame = ctx.new_frame();
+  ///
+  /// ctx.send_frame(frame)?;
+  /// ctx.flush();
+  ///
+  /// loop {
+  ///     match ctx.receive_packet() {
+  ///         Ok(packet) => { /* ... */ },
+  ///         Err(EncoderStatus::Encoded) => (),
+  ///         Err(EncoderStatus::LimitReached) => break,
+  ///         Err(err) => return Err(err),
+  ///     }
+  /// }
+  /// # Ok(())
+  /// # }
+  /// ```
+  #[inline]
   pub fn receive_packet(&mut self) -> Result<Packet<T>, EncoderStatus> {
     let inner = &mut self.inner;
     let pool = &mut self.pool;
 
     pool.install(|| inner.receive_packet())
   }
 
+  /// Flushes the encoder.
+  ///
+  /// Flushing signals the end of the video. After the encoder has been
+  /// flushed, no additional frames are accepted.
```

Comment:
Is this stuff true? I think @lu-zero wanted to make mid-stream flushing work.

Reply:
I do not have a strong opinion on this, supporting it shouldn't be much code but the documentation is matching the current status.

Reply:
If it matches current behavior, it's fine.

Reply:
I was under the impression that he PRd the opposite not too long ago: https://github.com/xiph/rav1e/pull/1392